cmake/cmake_refresh_git_revision.cmake

```diff
@@ -11,17 +11,19 @@
 #
 
 find_package(Git REQUIRED)
-set(preCICE_REVISION "no-info [Git failed/Not a repository]")
 execute_process(
   COMMAND ${GIT_EXECUTABLE} describe --tags --broken
```

Comment:
FYI, this does not work on Ubuntu 16.04, but it's fine as it should work for 18.04 (the oprion was added in between):

```
-- Found Git: /usr/bin/git (found version "2.7.4") 
error: unknown option `broken'
usage: git describe [<options>] [<commit-ish>...]
   or: git describe [<options>] --dirty

    --contains            find the tag that comes after the commit
    --debug               debug search strategy on stderr
    --all                 use any ref
    --tags                use any tag, even unannotated
    --long                always use long format
    --first-parent        only follow first parent
    --abbrev[=<n>]        use <n> digits to display SHA-1s
    --exact-match         only output exact matches
    --candidates <n>      consider <n> most recent tags (default: 10)
    --match <pattern>     only consider tags matching <pattern>
    --always              show abbreviated commit object as fallback
    --dirty[=<mark>]      append <mark> on dirty working tree (default: "-dirty")

-- Revision status: Detection failed
[  0%] Built target GitRevision
```

Reply:
As `--broken` is not necessary, I reverted to using `--dirty`. So the detection will now fail when the repo is broken.
I think that is a negligible corner case and well worth the compatibility with ubuntu 16.04.